pulsar-client-cpp/tests/BasicEndToEndTest.cc

```diff
@@ -578,8 +578,13 @@ TEST(BasicEndToEndTest, testSinglePartitionRoutingPolicy)
         consumer.receive(m);
         consumer.acknowledgeCumulative(m);
     }
+
+    Promise<bool, Result> closePromise;
+    Result r;
     consumer.close();
-    producer.closeAsync(0);
+    producer.closeAsync(WaitForCallback(closePromise));
```

Comment:
We can but closeAsync was used in an effort to increase code coverage of CPP Unit tests.